PWGCF/EbyEFluctuations/Tasks/Diff_pT_fluct_PID.cxx

```diff
@@ -0,0 +1,758 @@
+        histos.fill(HIST("QA/Pion/before/h2_TpcTofNsigma"), track.tpcNSigmaPi(), track.tofNSigmaPi());
+
+        histos.fill(HIST("QA/Proton/before/h2_TPCNsigma"), track.pt(), track.tpcNSigmaPr());
+        histos.fill(HIST("QA/Proton/before/h2_TOFNsigma"), track.pt(), track.tofNSigmaPr());
+        histos.fill(HIST("QA/Proton/before/h2_TpcTofNsigma"), track.tpcNSigmaPr(), track.tofNSigmaPr());
+
+        histos.fill(HIST("QA/Kaon/before/h2_TPCNsigma"), track.pt(), track.tpcNSigmaKa());
+        histos.fill(HIST("QA/Kaon/before/h2_TOFNsigma"), track.pt(), track.tofNSigmaKa());
+        histos.fill(HIST("QA/Kaon/before/h2_TpcTofNsigma"), track.tpcNSigmaKa(), track.tofNSigmaKa());
+
+        const float combNSigmaPi1 = std::sqrt(pow(track.tpcNSigmaPi(), 2.0) + pow(track.tofNSigmaPi(), 2.0));
+        const float combNSigmaKa1 = std::sqrt(pow(track.tpcNSigmaKa(), 2.0) + pow(track.tofNSigmaKa(), 2.0));
+        const float combNSigmaPr1 = std::sqrt(pow(track.tpcNSigmaPr(), 2.0) + pow(track.tofNSigmaPr(), 2.0));
+
+        histos.fill(HIST("QA/Pion/before/h2_TpcTofNsigma1"), track.pt(), combNSigmaPi1);
+        histos.fill(HIST("QA/Kaon/before/h2_TpcTofNsigma1"), track.pt(), combNSigmaKa1);
+        histos.fill(HIST("QA/Proton/before/h2_TpcTofNsigma1"), track.pt(), combNSigmaPr1);
+
+        if (track.eta() < 0) {
+          Nch_eta_less_0++;
+          pt_Ch = track.pt();
+
+          pT_sum_etaLess0 += pt_Ch;
+          fPt_profile->Fill(pt_Ch);
+
+          // For Pions:
+          if (selPions(track)) {
+            N_Pi_eta_less_0++;
+            pt_Pi = track.pt();
+
+            fPt_profile_pion->Fill(pt_Pi);
+
+            // QA
+            histos.fill(HIST("QA/Pion/h_Pt"), track.pt());
+            histos.fill(HIST("QA/Pion/h_rap"), track.rapidity(massPi));
+            histos.fill(HIST("QA/Pion/h2_Pt_rap"), track.rapidity(massPi), track.pt());
+            histos.fill(HIST("QA/Pion/h2_DcaXY"), track.pt(), track.dcaXY());
+            histos.fill(HIST("QA/Pion/h2_DcaZ"), track.pt(), track.dcaZ());
+
+            histos.fill(HIST("QA/Pion/h2_TPCNsigma"), track.pt(), track.tpcNSigmaPi());
+            histos.fill(HIST("QA/Pion/h2_TOFNsigma"), track.pt(), track.tofNSigmaPi());
+            histos.fill(HIST("QA/Pion/h2_TpcTofNsigma"), track.tpcNSigmaPi(), track.tofNSigmaPi());
+            histos.fill(HIST("QA/Pion/h2_TOFSignal"), track.pt(), track.beta());
+            histos.fill(HIST("QA/Pion/h2_TPCSignal"), track.pt(), track.tpcSignal());
+            histos.fill(HIST("QA/Pion/h2_ExpTPCSignal"), track.pt(), track.tpcExpSignalPi(track.tpcSignal()));
+
+            histos.fill(HIST("QA/after/h2_TOFSignal"), track.pt(), track.beta());
+            histos.fill(HIST("QA/after/h2_TPCSignal"), track.pt(), track.tpcSignal());
+
+            const float combNSigmaPi2 = std::sqrt(pow(track.tpcNSigmaPi(), 2.0) + pow(track.tofNSigmaPi(), 2.0));
+
+            histos.fill(HIST("QA/Pion/h2_TpcTofNsigma1"), track.pt(), combNSigmaPi2);
+          }
```

Comment:
You are basically replicating your code three times
Replicating code is error prone and difficult of maintaining it
I would recommend trying to structure your data in a was that can be used for the different species without the need of replicating code